django/otto/models.py

```diff
@@ -267,17 +267,41 @@ def __str__(self):
 
 
 class CostManager(models.Manager):
-    def new(self, user: User, feature: str, cost_type: str, count: int) -> "Cost":
-        feature = Feature.objects.get(short_name=feature)
+
+    def new(self, cost_type: str, count: int) -> "Cost":
+        from structlog.contextvars import get_contextvars
```

Comment:
This is the guts of the refactor. This method now only takes two required arguments. The rest are populated from contextvars, if available.